diff --git a/platforms/ios/build_framework.py b/platforms/ios/build_framework.py
index 4d4f7e3..b34922a 100755
--- a/platforms/ios/build_framework.py
+++ b/platforms/ios/build_framework.py
@@ -39,9 +39,10 @@ def build_opencv(srcroot, buildroot, target, arch):
     cmakeargs = ("-GXcode " +
                 "-DCMAKE_BUILD_TYPE=Release " +
                 "-DCMAKE_TOOLCHAIN_FILE=%s/platforms/ios/cmake/Toolchains/Toolchain-%s_Xcode.cmake " +
+                "-DOPENCV_EXTRA_MODULES_PATH=%s/../opencv_contrib/modules " +
                 "-DBUILD_opencv_world=ON " +
                 "-DCMAKE_C_FLAGS=\"-Wno-implicit-function-declaration\" " +
-                "-DCMAKE_INSTALL_PREFIX=install") % (srcroot, target)
+                "-DCMAKE_INSTALL_PREFIX=install") % (srcroot, target, srcroot)
     # if cmake cache exists, just rerun cmake to update OpenCV.xproj if necessary
     if os.path.isfile(os.path.join(builddir, "CMakeCache.txt")):
         os.system("cmake %s ." % (cmakeargs,))
